### PR TITLE
Added option for plinth --setup

### PR DESCRIPTION
**Pull request by harry-7**
Now plinth --setup could take a list of modules to setup. In reference #473

### CODE REVIEW

File: plinth/setup.py [hunk 1]
```diff
@@ -158,3 +158,12 @@ def setup_all_modules(essential=False):
             continue
 
         module.setup_helper.run()
+
+def setup_selected(module_list):
+    """Runs setup for selected modules and exit."""
+    logger.info('Running setup for Given modules')
```
Comment (SunilMohanAdapa) [inline plinth/setup.py]: 'Running setup for given modules'

File: plinth/setup.py [hunk 2]
```diff
@@ -158,3 +158,12 @@ def setup_all_modules(essential=False):
...
+    for module_name, module in plinth.module_loader.loaded_modules.items():
+        if module_name not in module_list:
+            continue
+
+        module.setup_helper.run()
```
Comment (SunilMohanAdapa) [inline plinth/setup.py]: Would you please consider merging the two setup methods?  We currently have one line to actually run the setup but that might get expanded later.
Something like
```
def setup_modules(modules_list=None, essential=False):
```

**Comment by harry-7** [comment]
@SunilMohanAdapa  I merged both methods and renamed the setup method as setup_modules instead of setup_all_modules Is that okay ?

File: plinth/setup.py [hunk 3]
```diff
@@ -150,11 +150,13 @@ def init(module_name, module):
         module.setup_helper = Helper(module_name, module)
 
 
-def setup_all_modules(essential=False):
+def setup_modules(module_list=None,essential=False):
```
Comment (SunilMohanAdapa) [inline plinth/setup.py]: Missing space after comma.

File: plinth/setup.py [hunk 4]
```diff
@@ -150,11 +150,13 @@ def init(module_name, module):
...
     """Run setup on all essential modules and exit."""
-    logger.info('Running setup for all modules, essential - %s', essential)
+    logger.info('Running setup for modules, essential - %s, selected modules - %s', essential,module_list)
```
Comment (SunilMohanAdapa) [inline plinth/setup.py]: Line greater than 79 characters.
Missing space after comma.

File: plinth/setup.py [hunk 5]
```diff
@@ -150,11 +150,13 @@ def init(module_name, module):
...
     for module_name, module in plinth.module_loader.loaded_modules.items():
         if essential and not getattr(module, 'is_essential', False):
             continue
+        if len(module_list)!=0 and module_name not in module_list:
```
Comment (SunilMohanAdapa) [inline plinth/setup.py]: Emptyness can be checked as `if module_list and...`

**Comment by SunilMohanAdapa** [comment]
The patch looks good now, I will make the minor corrections and commit it.  Is the .egg/README.txt commit intentional?

**Comment by SunilMohanAdapa** [comment]
You should also integrate flake8 and pylint tools to your editor (if you are on Emacs, try the flycheck mode) to avoid minor styling and other issues.

**Comment by harry-7** [comment]
@SunilMohanAdapa No that was not intentional Sorry . And Also I Use IDE Yeah I have but I did not make it. I will make a commit correcting them .

**Comment by SunilMohanAdapa** [comment]
I have rebased, squashed, signed and merged the patch with very minor changes.

Thank you for the patch.

**Pull request closed without merging**